resources/js/contact.js

```diff
@@ -0,0 +1,157 @@
+// https://youtu.be/eg4e-FObyJ8
+const submitBtn = document.querySelector(".formBtn")
+
+
+submitBtn.addEventListener('click', function(){
+  
+  const name = document.querySelector("#name").value
+  const subject = document.querySelector("#subject").value
+  const email = document.querySelector("#email").value
+  const phone = document.querySelector("#phone").value
+  const message = document.querySelector("#message").value
+  
+  if (validate(name, email, phone, message)){
+    var contactData = {name:name, email:email, phone:phone, message:message} 
+    const contactJSON = JSON.stringify(contactData)
+    // clearError()
+    apiCall(contactJSON)
+  }
+  else{
+    return
+  }
+  
+})
+
+
+
+function validate(name, email, phone, message){
+  // clearError();
+  var validName = false;
+  var validEmail = false;
+  var validPhone = false;
+  var validMessage = false;
+  console.log(name.length);
+
+  if (name.length == 0) {
+    Id1 = "#name-error";
```

Comment:
Select these elements and assign them to variables to avoid querying the DOM multiple times

Reply:
It has been moved to functions. Query will happen only once now.